src/dispatching/update_listeners/polling.rs

```diff
@@ -1,89 +1,192 @@
-use std::{convert::TryInto, time::Duration};
-
-use futures::{
-    future::{ready, Either},
-    stream::{self, Stream, StreamExt},
+use std::{
+    convert::TryInto,
+    future::Future,
+    pin::Pin,
+    task::{
+        self,
+        Poll::{self, Ready},
+    },
+    time::Duration,
+    vec,
 };
 
+use futures::{ready, stream::Stream};
+
 use crate::{
     dispatching::{
         stop_token::{AsyncStopFlag, AsyncStopToken},
-        update_listeners::{stateful_listener::StatefulListener, UpdateListener},
+        update_listeners::{AsUpdateStream, UpdateListener},
     },
-    payloads::{GetUpdates, GetUpdatesSetters as _},
     requests::{HasPayload, Request, Requester},
     types::{AllowedUpdate, Update},
 };
 
+/// Builder for polling update listener.
+///
+/// Can be created by [`Polling::builder`].
+pub struct PollingBuilder<R> {
+    bot: R,
+    timeout: Option<Duration>,
+    limit: Option<u8>,
+    allowed_updates: Option<Vec<AllowedUpdate>>,
+    drop_pending_updates: bool,
+}
+
+impl<R> PollingBuilder<R>
+where
+    R: Requester + Send + 'static,
+    <R as Requester>::GetUpdates: Send,
+{
+    /// A timeout in seconds for polling.
+    ///
+    /// ## Note
+    ///
+    /// `timeout` should not be bigger than http client timeout, see
+    /// [`default_reqwest_settings`] for default http client settings.
+    ///
+    /// [`default_reqwest_settings`]: crate::net::default_reqwest_settings
+    pub fn timeout(self, timeout: Duration) -> Self {
```

Comment:
How about adding `#[must_use]` to these functions?

Reply:
Done 50f9d3b46e117c543a3eb1061b11b8a22f4e954a